Problem with release build on Windows

I am trying to do a release build to run some benchmarks `cargo build --release`. debug build works fine but release fails. Some of the patches are failing and then there is lots of compile errors (I assume because the patches fail).

I am on Windows and using Rust 1.56. 

Any thoughts on the issue? I can see in the build script there are references to "TARGET" for bindgen fn but not sure of the purpose.

```
Caused by:
  process didn't exit successfully: `C:\Users\tryan\.cargo\registry\src\github.com-1ecc6299db9ec823\rquickjs-sys-0.1.3\target\release\build\rquickjs-sys-ff47a7935ac38437\build-script-build` (exit code: 1)
  --- stdout
  cargo:rerun-if-changed=build.rs
  cargo:rerun-if-env-changed=CARGO_FEATURE_EXPORTS
  cargo:rerun-if-env-changed=CARGO_FEATURE_BINDGEN
  cargo:rerun-if-env-changed=CARGO_FEATURE_UPDATE_BINDINGS
  cargo:rerun-if-env-changed=CARGO_FEATURE_DUMP_BYTECODE
  cargo:rerun-if-env-changed=CARGO_FEATURE_DUMP_GC
  cargo:rerun-if-env-changed=CARGO_FEATURE_DUMP_GC_FREE
  cargo:rerun-if-env-changed=CARGO_FEATURE_DUMP_FREE
  cargo:rerun-if-env-changed=CARGO_FEATURE_DUMP_LEAKS
  cargo:rerun-if-env-changed=CARGO_FEATURE_DUMP_MEM
  cargo:rerun-if-env-changed=CARGO_FEATURE_DUMP_OBJECTS
  cargo:rerun-if-env-changed=CARGO_FEATURE_DUMP_ATOMS
  cargo:rerun-if-env-changed=CARGO_FEATURE_DUMP_SHAPES
  cargo:rerun-if-env-changed=CARGO_FEATURE_DUMP_MODULE_RESOLVE
  cargo:rerun-if-env-changed=CARGO_FEATURE_DUMP_PROMISE
  cargo:rerun-if-env-changed=CARGO_FEATURE_DUMP_READ_OBJECT
  Applyng patch patches\check_stack_overflow.patch
  patching file quickjs.c
  Hunk #1 FAILED at 1587.
  1 out of 1 hunk FAILED -- saving rejects to file quickjs.c.rej
  Applyng patch patches\infinity_handling.patch
  patching file quickjs.c
  Hunk #1 succeeded at 10237 (offset 46 lines).
  Hunk #2 succeeded at 41734 (offset 170 lines).
  Hunk #3 succeeded at 47841 (offset 170 lines).
  Applyng patch patches\atomic_new_class_id.patch
  patching file quickjs.c
  Hunk #1 succeeded at 3373 (offset 20 lines).
  Applyng patch patches\basic_msvc_compat.patch
  patching file cutils.c
  patching file cutils.h
  patching file libbf.h
  patching file quickjs.c
  Hunk #2 succeeded at 49 (offset 2 lines).
  Hunk #3 succeeded at 68 (offset 2 lines).
  Hunk #4 succeeded at 7247 (offset 22 lines).
  Hunk #5 succeeded at 7278 (offset 22 lines).
  Hunk #6 succeeded at 7305 (offset 22 lines).
  Hunk #7 succeeded at 7395 (offset 22 lines).
  Hunk #8 succeeded at 9047 (offset 46 lines).
  Hunk #9 succeeded at 16048 (offset 95 lines).
  Hunk #10 succeeded at 16061 (offset 95 lines).
  Hunk #11 succeeded at 16292 (offset 95 lines).
  Hunk #12 succeeded at 39263 (offset 170 lines).
  Hunk #13 succeeded at 40681 (offset 170 lines).
  Hunk #14 succeeded at 41994 with fuzz 1 (offset 170 lines).
  Hunk #15 succeeded at 45711 (offset 170 lines).
  Hunk #16 succeeded at 45933 (offset 170 lines).
  Hunk #17 succeeded at 46911 (offset 170 lines).
  Hunk #18 succeeded at 47270 (offset 170 lines).
  Hunk #19 succeeded at 52699 (offset 236 lines).
  patching file quickjs.h
  Hunk #1 succeeded at 666 (offset 4 lines).
  Hunk #2 succeeded at 675 (offset 4 lines).
  TARGET = Some("x86_64-pc-windows-msvc")
  OPT_LEVEL = Some("3")
  HOST = Some("x86_64-pc-windows-msvc")
  CC_x86_64-pc-windows-msvc = None
  CC_x86_64_pc_windows_msvc = None
  HOST_CC = None
  CC = None
  CFLAGS_x86_64-pc-windows-msvc = None
  CFLAGS_x86_64_pc_windows_msvc = None
  HOST_CFLAGS = None
  CFLAGS = None
  CRATE_CC_NO_DEFAULTS = None
  CARGO_CFG_TARGET_FEATURE = Some("fxsr,sse,sse2")
  DEBUG = Some("false")
  running: "C:\\Program Files (x86)\\Microsoft Visual Studio\\2017\\BuildTools\\VC\\Tools\\MSVC\\14.16.27023\\bin\\HostX64\\x64\\cl.exe" "-nologo" "-MD" "-O2" "-Brepro" "-W4" "-D_GNU_SOURCE" "-DCONFIG_VERSION=\"2020-01-19\"" "-DCONFIG_BIGNUM" "-FoC:\\Users\\tryan\\.cargo\\registry\\src\\github.com-1ecc6299db9ec823\\rquickjs-sys-0.1.3\\target\\release\\build\\rquickjs-sys-5860fc66d775156b\\out\\libregexp.o" "-c" "C:\\Users\\tryan\\.cargo\\registry\\src\\github.com-1ecc6299db9ec823\\rquickjs-sys-0.1.3\\target\\release\\build\\rquickjs-sys-5860fc66d775156b\\out\\libregexp.c"
  libregexp.c
  C:\Users\tryan\.cargo\registry\src\github.com-1ecc6299db9ec823\rquickjs-sys-0.1.3\target\release\build\rquickjs-sys-5860fc66d775156b\out\libregexp.c(264): warning C4245: '+=': conversion from 'int' to 'uint32_t', signed/unsigned mismatch
  C:\Users\tryan\.cargo\registry\src\github.com-1ecc6299db9ec823\rquickjs-sys-0.1.3\target\release\build\rquickjs-sys-5860fc66d775156b\out\libregexp.c(396): warning C4244: 'function': conversion from 'int' to 'uint8_t', possible loss of data
  C:\Users\tryan\.cargo\registry\src\github.com-1ecc6299db9ec823\rquickjs-sys-0.1.3\target\release\build\rquickjs-sys-5860fc66d775156b\out\libregexp.c(403): warning C4244: 'function': conversion from 'int' to 'uint8_t', possible loss of data
  C:\Users\tryan\.cargo\registry\src\github.com-1ecc6299db9ec823\rquickjs-sys-0.1.3\target\release\build\rquickjs-sys-5860fc66d775156b\out\libregexp.c(404): warning C4267: '=': conversion from 'size_t' to 'int', possible loss of data
  C:\Users\tryan\.cargo\registry\src\github.com-1ecc6299db9ec823\rquickjs-sys-0.1.3\target\release\build\rquickjs-sys-5860fc66d775156b\out\libregexp.c(412): warning C4244: 'function': conversion from 'int' to 'uint8_t', possible loss of data
  C:\Users\tryan\.cargo\registry\src\github.com-1ecc6299db9ec823\rquickjs-sys-0.1.3\target\release\build\rquickjs-sys-5860fc66d775156b\out\libregexp.c(413): warning C4267: '=': conversion from 'size_t' to 'int', possible loss of data
  C:\Users\tryan\.cargo\registry\src\github.com-1ecc6299db9ec823\rquickjs-sys-0.1.3\target\release\build\rquickjs-sys-5860fc66d775156b\out\libregexp.c(420): warning C4244: 'function': conversion from 'int' to 'uint8_t', possible loss of data
  C:\Users\tryan\.cargo\registry\src\github.com-1ecc6299db9ec823\rquickjs-sys-0.1.3\target\release\build\rquickjs-sys-5860fc66d775156b\out\libregexp.c(421): warning C4244: 'function': conversion from 'uint32_t' to 'uint8_t', possible loss of data
``

in my case all of a sudden it failed to build either debug or release on windows with following error

```
target\debug\build\rquickjs-sys-45074756569ed1f5\out\quickjs.c(31): fatal error C1083: Cannot open include file: 'sys/time.h': No such file or directory
```

I can't post it all as there is too much text, 2000+ lines. I can find to lines that are errors:

```
  D:\Documents\Projects\src\aaaa\rquickjs\target\release\build\rquickjs-sys-d0f71851adcfbdf8\out\quickjs.c(41952): error C2099: initializer is not a constant
  D:\Documents\Projects\src\aaaa\rquickjs\target\release\build\rquickjs-sys-d0f71851adcfbdf8\out\quickjs.c(41953): error C2099: initializer is not a constant
```

It looks like the issue is on the floor and ceil functions based on the line numbers:
```
    JS_CFUNC_SPECIAL_DEF("floor", 1, f_f, floor ),
    JS_CFUNC_SPECIAL_DEF("ceil", 1, f_f, ceil ),
```

--- a/sys/patches/basic_msvc_compat.patch
+++ b/sys/patches/basic_msvc_compat.patch
@@ -392,3 +392,17 @@ index 92457a2..f60a937 100644
  }
  
  int JS_ToBool(JSContext *ctx, JSValueConst val); /* return -1 for JS_EXCEPTION */
+diff --git a/quickjs.c b/quickjs.c
+index 48aeffc..4ac7507 100644
+--- a/quickjs.c
++++ b/quickjs.c
+@@ -41906,6 +41906,9 @@ static JSValue js_math_random(JSContext *ctx, JSValueConst this_val,
+     return __JS_NewFloat64(ctx, u.d - 1.0);
+ }
+
++#pragma function (ceil)
++#pragma function (floor)
++
+ static const JSCFunctionListEntry js_math_funcs[] = {
+     JS_CFUNC_MAGIC_DEF("min", 2, js_math_min_max, 0 ),
+     JS_CFUNC_MAGIC_DEF("max", 2, js_math_min_max, 1 ),

